org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/serializer/impl/ChangeSerializer.java

```diff
@@ -153,14 +181,16 @@ protected void endRecordChanges(IAcceptor<IEmfResourceChange> changeAcceptor) {
 				updaters.add(updater);
 			}
 		}
-		SubMonitor progress = SubMonitor.convert(monitor, "Creating Text Changes...", updaters.size());
+		
+		monitor.beginTask("Creating text changes...", updaters.size());
 		for (ResourceUpdater updater : updaters) {
+			monitor.split(1);
 			updater.applyChange(deltas, changeAcceptor);
-			progress.split(1);
 		}
 		for (ResourceUpdater updater : updaters) {
 			updater.unload();
 		}
+		monitor.done();
```

Comment:
When you use a dedicated SubMonitor within the method, `done` does not need to be called.

Reply:
If you have a look into the impl of `split()`, you'll notice that split just "reserves"  work in the parent monitor, but doesn't report it via `worked()`. In stead it reports the work of the formerly split child monitor and its children. So in order to get the last piece of work reported, too, `done()` is called here. According to https://www.eclipse.org/articles/Article-Progress-Monitors/article.html, Section `4.0 Responsibilities of callers and callees` this is valid.